lib/internal/modules/esm/hooks.js

```diff
@@ -553,15 +558,26 @@ class HooksProxy {
     }
   }
 
-  async makeAsyncRequest(method, ...args) {
+  /**
+   * Invoke a remote method asynchronously.
+   * @param {any[]} transferList Objects in `args` to be transferred
+   * @param {string} method Method to invoke
+   * @param  {any[]} args Arguments to pass to `method`
+   * @returns {Promise<any>}
+   */
+  async makeAsyncRequest(transferList, method, ...args) {
     this.waitForWorker();
 
     MessageChannel ??= require('internal/worker/io').MessageChannel;
     const asyncCommChannel = new MessageChannel();
 
     // Pass work to the worker.
-    debug('post async message to worker', { method, args });
-    this.#worker.postMessage({ method, args, port: asyncCommChannel.port2 }, [asyncCommChannel.port2]);
+    debug('post async message to worker', { method, args, transferList });
+    const finalTransferList = [asyncCommChannel.port2];
+    if (transferList) {
+      ArrayPrototypePushApply(finalTransferList, transferList);
+    }
```

Comment:
I'm pretty sure this wouldn't work, it would throw something like `Value at index 1 does not have a transferable type.`

Reply:
🤯 why?

Reply:
Because arrays are not transferable (see https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects#supported_objects). You can convince yourself by running e.g. `{const a=new ArrayBuffer(9);structuredClone({a}, {transfer:[[a]]})}`.

Reply:
Ohhh! Whoops this is `push.apply`, not `push`. Nvmd, they're not equivalent.